src/telemetry/main.cpp

```diff
@@ -25,56 +25,100 @@
 
 namespace hyped {
 
-using hyped::client::Client;
+using client::Client;
+using data::ModuleStatus;
+using data::Data;
 
 namespace telemetry {
 
 Main::Main(uint8_t id, Logger& log)
     : Thread(id, log),
-      client_ {log}
+      client_ {log},
+      data_ {Data::getInstance()}
 {
     log_.DBG("Telemetry", "Telemetry thread started");
 }
 
 void Main::run()
 {
-    std::thread recvThread {recvLoop, std::ref(client_)};  // NOLINT (linter thinks semicolon is syntax error...)
-
-    while (true) {
-        protoTypes::TestMessage msg;
+    if (!client_.connect()) {
+        // idk throw exception or something
+        log_.ERR("Telemetry", "ERROR CONNECTING TO SERVER");
+    }
 
-        msg.set_command(protoTypes::TestMessage::VELOCITY);
-        msg.set_data(222);
-        client_.sendData(msg);
+    // syntax explanation so I don't forget: thread constructor expects pointer to member function,
+    //                                       also needs 'this' as object to call member function on
+    std::thread recv_thread {&Main::recvLoop, this};  // NOLINT (linter thinks semicolon is syntax error...)
```

Comment:
do not use `std:thread`, use our `hyped::utils::concurrent::Thread` abstraction

Reply:
Will include in next PR